src/core/utils/layerutils.cpp

```diff
@@ -203,3 +204,45 @@ bool LayerUtils::deleteFeature( QgsProject *project, QgsVectorLayer *layer, cons
 
   return isSuccess;
 }
+
+QgsFeature LayerUtils::duplicateFeature( QgsVectorLayer *layer, const QgsFeature &feature, bool toggleEditing )
+{
+  if ( !layer )
+  {
+    QgsMessageLog::logMessage( tr( "Cannot start editing, no layer" ), "QField", Qgis::Warning );
+    return QgsFeature();
+  }
+
+  if ( toggleEditing && ( !layer->startEditing() || !layer->editBuffer() ) )
+  {
+    QgsMessageLog::logMessage( tr( "Cannot start editing" ), "QField", Qgis::Warning );
+    return QgsFeature();
+  }
+
+  QgsFeature duplicatedFeature;
+  QObject tmp;
+  auto featureAdded = [ layer, &duplicatedFeature ]( QgsFeatureId fid )
+  {
+    duplicatedFeature = layer->getFeature( fid );
+  };
+  connect( layer, &QgsVectorLayer::featureAdded, &tmp, featureAdded );
```

Comment:
```suggestion
  auto featureAdded = [ layer, &duplicatedFeature ]( QgsFeatureId fid )
  {
    duplicatedFeature = layer->getFeature( fid );
  };
  QMetaObject::Connection connection = connect( layer, &QgsVectorLayer::featureAdded, featureAdded );
  auto sweaper = qScopeGuard([layer, connection] { layer->disconnect( connection ); });
```

just because

Reply:
@m-kuhn , not just because, that's what I'd call doing the right thing :)